http/src/main/scala/org/http4s/blaze/http/http2/StreamState.scala

```diff
@@ -4,13 +4,38 @@ import java.nio.ByteBuffer
 import java.util
 
 import org.http4s.blaze.http.http2.Http2Exception._
-import org.http4s.blaze.pipeline.Command
 import org.http4s.blaze.pipeline.Command.{EOF, OutboundCommand}
-import org.http4s.blaze.pipeline.HeadStage
+import org.http4s.blaze.pipeline.{Command, HeadStage}
 import org.http4s.blaze.util.BufferTools
 
 import scala.concurrent.{Future, Promise}
 
+private trait StreamState extends HeadStage[StreamMessage] with WriteInterest {
+
+  /** Stream ID associated with this stream */
+  def streamId: Int
+
+  /** The flow window associated with this stream */
+  def flowWindow: StreamFlowWindow
+
+  /** Called when the outbound flow window of the session or this stream has had some data
+    * acked and we may now be able to make forward progress.
+    */
+  def outboundFlowWindowChanged(): Unit
+
+  /** Must be called by the [[WriteController]] from within the session executor */
+  def performStreamWrite(): Seq[ByteBuffer]
+
+  /** Called by the session when a DATA frame has been received from the remote peer */
+  def invokeInboundData(endStream: Boolean, data: ByteBuffer, flowBytes: Int): MaybeError
+
+  /** Called by the session when a HEADERS has been received from the remote peer */
+  def invokeInboundHeaders(priority: Priority, endStream: Boolean, headers: Seq[(String,String)]): MaybeError
+
+  /** Close the stream, optionally with an error */
+  def closeWithError(t: Option[Throwable]): Unit
```

Comment:
Similar question to the other close Unit.

Reply:
Roughly the same answer: I'm need this to complete before I move on to maintain coherence of the session state. Eg, if I close a stream, I need to know that it is gone *now*.